src/OfflineTool.cpp

```diff
@@ -0,0 +1,473 @@
+
+}
+
+std::string
+getStdin()
+{
+    std::ostringstream stdinput;
+    stdinput << std::cin.rdbuf();
+    return stdinput.str();
+}
+
+void
+runCommand (const std::string& command,
+    std::vector <std::string> const& args,
+    boost::filesystem::path const& keyFile,
+    boost::optional<std::string> const& keyType,
+    boost::optional<std::string> const& seed)
+{
+    using namespace std;
+
+    struct commandParams
+    {
+        vector<string>::size_type minParams;
+        vector<string>::size_type maxParams;
+        bool enableStdin;
+    };
+    static map<string, commandParams>
+        const commandArgs = {
+            { "serialize", {0, 1, true} },
+            { "deserialize", {0, 1, true} },
+            { "sign", {0, 1, true} },
+            { "multi_sign", {0, 1, true} },
+            { "create_keyfile", {0, 0, false} },
+    };
+
+    auto const iArgs = commandArgs.find(command);
+
+    if (iArgs == commandArgs.end())
+        throw std::runtime_error("Unknown command: " + command);
+
+    if (args.size() < iArgs->second.minParams ||
+            args.size() > iArgs->second.maxParams)
+        throw std::runtime_error("Syntax error: Wrong number of arguments");
+
+    auto const input = [&]
+    {
+        if (args.size() == 0)
+            return iArgs->second.enableStdin ? getStdin() : string{};
+        else
+            return args[0];
+    }();
+
+    if (command == "serialize")
+        doSerialize(input);
+    else if (command == "deserialize")
+        doDeserialize(input);
+    else if (command == "sign")
+        doSingleSign(input, keyFile);
+    else if (command == "multi_sign")
+        doMultiSign(input, keyFile);
+    else if (command == "create_keyfile")
+        doCreateKeyfile(keyFile, keyType, seed);
+}
+
+//LCOV_EXCL_START
+static
+std::string
+getEnvVar (char const* name)
+{
+    std::string value;
+
+    auto const v = getenv (name);
+
+    if (v != nullptr)
+        value = v;
+
+    return value;
+}
+
+void printHelp (const boost::program_options::options_description& desc,
+    boost::filesystem::path const& defaultKeyfile)
+{
+    static std::string const name = "ripple-offline-tool";
+
+    std::cerr
+        << name << " [options] <command> [<argument> ...]\n"
+        << desc << std::endl <<
+R"(Commands:
+  Serialization:
+    serialize [<argument>]              Serialize from JSON.
+    deserialize [<argument>]            Deserialize to JSON.
+  Transaction signing:
+    sign [<argument>]                   Sign for submission.
+    multi_sign [<argument>]             Apply a multi-signature.
+      Signing commands require a valid keyfile.
+      Input is serialized or unserialized JSON.
+      Output is unserialized JSON.
+  For serialization and signing commands, if an <argument>
+      is not provided, the data is read from stdin.
+  Key Management:
+    create_keyfile                      Create keyfile using the options above.
+
+      Default keyfile is: )" << defaultKeyfile << "\n";
```

Comment:
While I was playing around with the tool I noticed one very minor oddness.  The name of the tool and the name of the key file use hyphens.  The command arguments, on the other hand, use underscores.  It's not essential that they use the same separator, but consider tidying that up before releasing the tool on the public.